### PR TITLE
ePIC: Add Blockminer support

### PR DESCRIPTION
**Pull request by jpcomps**
Adding as a draft pending review. If all looks good we can remove the draft status

Adds ePIC Blockminer support. @b-rowan you can comment if you like the hierarchy here. I know things get a bit complicated when model names are numbers (520i), so it created a situation with a double blockminer path. Let me know if you have any suggestions

One last change, added data_retries to the web-api to help with high latency situations (or when the UMC responds that the API was not ready)

### CODE REVIEW

File: pyasic/miners/models/epic/blockminer/blockminer.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from pyasic.miners.makes import ePICMake
+
+
+class Blockminer520(ePICMake):
+    raw_model = "Blockminer 520i"
```
Comment (b-rowan) [inline pyasic/miners/models/epic/blockminer/blockminer.py:5]: Any reason to not name this `Blockminer520i` instead?

Comment (jpcomps) [reply pyasic/miners/models/epic/blockminer/blockminer.py:5]: None other than me missing the i...good catch

File: pyasic/miners/blockminer/epic/blockminer/blockminer.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.    -
+#  See the License for the specific language governing permissions and         -
+#  limitations under the License.                                              -
+# ------------------------------------------------------------------------------
+
+from pyasic.miners.backends import ePIC
+from pyasic.miners.models import Blockminer520i
+
+
+class ePICBlockMiner520(ePIC, Blockminer520i):
```
Comment (b-rowan) [inline pyasic/miners/blockminer/epic/blockminer/blockminer.py:21]: Same here, missing an `i`.

File: pyasic/miners/models/epic/blockminer/blockminer.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from pyasic.miners.makes import ePICMake
+
+
+class Blockminer520i(ePICMake):
+    raw_model = "Blockminer 520i"
```
Comment (b-rowan) [inline pyasic/miners/models/epic/blockminer/blockminer.py:5]: Tempted to call this `BlockMiner` instead, you can compare to other miner types and decide, but it should be consistent with the subclass...

Comment (jpcomps) [reply pyasic/miners/models/epic/blockminer/blockminer.py:5]: I'm indifferent and can change it to what you prefer.. If I said there was a "Blockminer 720i" and then a follow on "Blockminer2 XXX" does that change the thought process?

Comment (UpstreamData) [reply pyasic/miners/models/epic/blockminer/blockminer.py:5]: I'm not sure we have a precedent for that yet.  Could make things complex...

I'm not opposed to having a `BlockMiner720i` and `BlockMiner2720i`, as long as they're distinct.

Even `BlockMiner1720i` isn't the worst thing, but I would much prefer to follow the API output/correct model information as much as possible.

I think honestly the manufacturer should just name them differently in the first place ;), and if I had my say on that one, I would actually have the 720i and the 1720i, use the last 4 characters as a sub model and the first one as a major version, similar to what Whatsminer does.

**Review by b-rowan — APPROVED** [review]

**Review by b-rowan — APPROVED** [review]

**Pull request merged**